SweetTweetApp/src/main/webapp/resourses/js/App.js

```diff
@@ -170,30 +170,186 @@ const POSTS = [
+        });
+    }
+
+    setPostForDeleting(postId) {
+        this.postForDeleting = postId;
+        this.message.classList.add("emergence");
+    }
+
+    setErrorField() {
+        this.error.classList.add("emergence");
+        this.closeButton = document.querySelector(".button-return");
+        this.closeButton.addEventListener("click", () => {
+                this.error.classList.remove("emergence");
+                this.setPage("posts")
+            }
+        );
+    }
+
+    isFull() {
+        return this.top >= (this.tweetList._availableId - 1);
+    }
+
+    renderPage() {
+        switch (this.currentPage) {
+            case "posts": {
+                this.renderPosts();
+                break;
+            }
+            case "authorization": {
+                this.renderAuthPage();
+                break;
+            }
+            case "addEditPage": {
+                this.renderAddEditPage();
+                break;
+            }
+            case "filtering": {
+                this.renderFilteringPosts();
+                break;
+            }
+        }
     }
 
     renderPosts() {
-        let view = new View();
-        view.addPostsToNewsLine(this.tweetList.getPosts(), "Alexander T");
+        this.viewPosts = this.tweetList.getPosts(0, this.top, this.filters);
+        if (this.viewPosts === "Incorrect data") {
+            this.renderError(this.viewPosts);
+        }
+        this.view.renderPostsPage(this.viewPosts, this.currentUser, this.setPostForDeleting.bind(this),
+            this.setOperation.bind(this), this.setPage.bind(this), this.login.bind(this), this.workWithLike.bind(this),
+            this.setButtonLoad.bind(this));
+        if (this.viewPosts.length === 0 || this.viewPosts.length === this.tweetList._posts.length) {
+            this.loadMore.classList.add("shadow");
+        }
+        this.filtering = document.querySelector(".filtering-button");
+        this.filtering.addEventListener("click", () => this.setPage("filtering"));
+    }
+
+    renderAuthPage() {
+        this.top = 10;
+        this.view.renderAuthorizationPage(this.setPage.bind(this), this.login.bind(this), this.renderError.bind(this));
+    }
+
+    renderAddEditPage() {
+        if (this.operation === "add") {
+            this.top = 10;
+            this.view.renderAddEditPage(this.currentUser, this.operation, {}, this.tweetList._availableId, this.addPost.bind(this));
+        } else {
+            this.top = 10;
+            this.view.renderAddEditPage(this.currentUser, this.operation, this.tweetList.getPost(this.operation), null, this.editPost.bind(this));
+        }
     }
 
-    addPost(post){
-        if(this.tweetList.addPost(post)){
-            this.renderPosts();
+    renderFilteringPosts() {
+        this.view.renderFilteringPosts(this.setFilters.bind(this));
+    }
+
+    renderError(textError) {
+        this.view.renderError(textError, this.setErrorField.bind(this));
+    }
+
+    addPost(post) {
```

Comment:
After adding post, it looks like it has wrong id in controller. While trying to like/edit/delete it, action is applied to the next one post

Reply:
Maybe it's look like that but when I checked everything was OK.

Reply:
Okay, I've rechecked it and it looks fine. IDK what was the reason on my local machine